crates/rethnet_evm/benches/state.rs

```diff
@@ -16,10 +19,19 @@ impl RethnetStates {
         let mut states: [&mut dyn SyncState<StateError>; 2] = [&mut self.layered, &mut self.hybrid];
         for state in states.iter_mut() {
             let number_of_checkpoints = number_of_accounts / number_of_accounts_per_checkpoint;
-            for _ in 0..=number_of_checkpoints {
-                for i in 1..=number_of_accounts_per_checkpoint {
+            for checkpoint_number in 0..=number_of_checkpoints {
+                for account_number in 1..=number_of_accounts_per_checkpoint {
+                    let address =
+                        (checkpoint_number * number_of_accounts_per_checkpoint) + account_number;
                     state
-                        .insert_account(Address::from_low_u64_ne(i), AccountInfo::default())
+                        .insert_account(
+                            Address::from_low_u64_ne(address),
+                            AccountInfo::new(
+                                U256::from(address),
+                                address,
+                                Bytecode::new_raw(Bytes::from(format!("0x{address:x}"))),
```

Comment:
address is a byte hash, so you can literally use those bytes directly.

Reply:
I'm not sure what you have in mind here. You're talking about line 32, right?

Using `address` directly for this parameter doesn't work because in this context it's a `u64`: ```expected struct `Bytecode`, found `u64` ```

but it seems like you're talking about an `Address`, which i understand is a byte hash, but using `Address::from_low_u64_ne(address)` for this parameter doesn't work either: ```expected struct `Bytecode`, found struct `B160` ```

i also tried using `Bytecode::new_raw(Address::from_low_u64_ne(address))` but that didn't work either: ```expected struct `rethnet_eth::Bytes`, found struct `B160` ```

are you perhaps confusing `code` with `code_hash` and thinking this argument is expecting a hash?

Reply:
`Address` is just a 20-byte static array. `Bytecode` can be constructed from a byte slice. So - one way of avoiding the string generation, it being the most expensive way of converting - you can generate the address from the u64 and then grab the byte slice of it to instantiate the Bytecode

Reply:
I think I achieved what you wanted (thought it didn't work out _exactly_ as you suggested) in https://github.com/NomicFoundation/hardhat/pull/3881/commits/2074b056344a742c870a3afd12c4efd5b9c8a559. Let me know what you think.